text/src/swam/witx/unresolved/Types.scala

```diff
@@ -0,0 +1,36 @@
+package swam
+package witx
+package unresolved
+
+/**
+@author Javier Cabrera-Arteaga on 2020-03-18
+  */
+sealed trait BaseWitxType
+
+case class BasicType(name: String) extends BaseWitxType
+
+case class AliasType(name: String) extends BaseWitxType
+
+case class EnumType(`type`: BaseWitxType, names: Seq[String]) extends BaseWitxType
```

Comment:
I generally tend to prefer using `tpe` instead of escaping the `type` keyword.

Reply:
Gotcha